src/adapted/mod.rs

```diff
@@ -257,8 +258,19 @@ impl<'a> CustomTryFrom<raw::client_messages::WriteEventsCompleted<'a>> for Adapt
                     //  * separate (instead of newtype for tuple)
                     // these must be options, as for idempotent writes the positions might not be
                     // returned. both seem to be returned always.
-                    prepare_position: msg.prepare_position.map(|x| x.into()),
-                    commit_position: msg.commit_position.map(|x| x.into()),
+
+                    // FIXME once this sort of error return were removed,
+                    // it will be possible to use the Carrier syntax (?)
```

Comment:
Yes, I can see how that could work..

```
impl<'a, 'b: 'a> TryFrom<&SomeProtocolType<'b>> for SomeAdaptedType<'a> {
}
```

(or `'a: 'b`, I can never get these right without the compiler)

This would definetly allow for the use of `?` in these converters. Though a third lifetime would need to be defined for the `&SomeProtocolType<'b>` borrow and I cannot figure out how it's bounds would go. Probably less than both `'a` and `'b`?

Rewriting these conversions is not currently a high priority for me as I'm still a bit skeptical if this `raw` and `adapted` even makes any sense compared alternatives, like providing extension traits for interesting  `RawMessage` responses.